### PR TITLE
add maxLen and stripTerminator args to terminated Transaction

### PR DESCRIPTION
**Pull request by EParisot**

### CODE REVIEW

**Review by altera2015 — APPROVED** [review]

**Comment by EParisot** [comment]
Ok I think I should do it again but from the new commit after previous PR and rework a bit

**Comment by EParisot** [comment]
ok this should be better, I added the optional arge on both Transformers also.
Can you re-run the merge ? or should I create another PR ?

**Comment by altera2015** [comment]
> ok this should be better, I added the optional arge on both Transformers also. Can you re-run the merge ? or should I create another PR ?

Hmm still failing some flutter checks :(

**Comment by EParisot** [comment]
Ican assume the semicolumn issues have been introduced by my PR but I don't know about the rest thought, did the PR 77 passed the same tests ?

**Comment by EParisot** [comment]
Also, I cant manage to make it works with my code, app crash as soon as I plug or unplug my usb device.
SDK 33 seems to have a bug regarding the getParcelableExtra : https://issuetracker.google.com/issues/240585930
I will have to revert to previous version on my fork for now...

**Comment by altera2015** [comment]
Not sure what is happening here. The upgrade to dart 3 likely caused these issues. I don't have time to dive in but this article does explain how to get passed them:

https://medium.com/@reginfell/flutter-dart-strong-mode-fd936bb8551f

**Comment by altera2015** [comment]
I'll revert the previous MR.

**Comment by altera2015** [comment]
I've reverted 77 and fixed the warnings, rebase on master.

**Comment by EParisot** [comment]
Yes multiple problems I think, I did the same on my fork thanks

**Comment by EParisot** [comment]
I just fixed the (formating) conflics, this works on my side

File: lib/transaction.dart [hunk 1]
```diff
@@ -40,17 +40,29 @@ class Transaction<T> {
   /// ```dart
   /// var c = Transaction.terminated(p.inputStream, Uint8List.fromList([13, 10]));
   /// ```
-  static Transaction<Uint8List> terminated(Stream<Uint8List> stream, Uint8List terminator) {
-    return Transaction<Uint8List>(stream, TerminatedTransformer.broadcast(terminator: terminator));
+  static Transaction<Uint8List> terminated(
+      Stream<Uint8List> stream, Uint8List terminator,
+      {int maxLen: 1024, bool stripTerminator: true}) {
```
Comment (altera2015) [inline lib/transaction.dart:45]:
```suggestion
      {int maxLen = 1024, bool stripTerminator = true}) {
```

File: lib/transaction.dart [hunk 2]
```diff
@@ -40,17 +40,29 @@ class Transaction<T> {
...
   /// Create a transaction that uses MagicHeaderAndLengthByteTransformer
   ///
   /// ```dart
   /// Transaction.magicHeader(p.inputStream, Uint8List.fromList([65,65,65])); // expects magic header AAA and then byte of length.
   /// ```
-  static Transaction<Uint8List> magicHeader(Stream<Uint8List> stream, List<int> header) {
-    return Transaction<Uint8List>(stream, MagicHeaderAndLengthByteTransformer.broadcast(header: header));
+  static Transaction<Uint8List> magicHeader(
+      Stream<Uint8List> stream, List<int> header,
+      {int maxLen: 1024}) {
```
Comment (altera2015) [inline lib/transaction.dart:61]:
```suggestion
      {int maxLen = 1024}) {
```

File: lib/transaction.dart [hunk 3]
```diff
@@ -59,13 +71,21 @@ class Transaction<T> {
   /// ```dart
   /// var c = Transaction.stringTerminated(p.inputStream, Uint8List.fromList([13, 10]));
   /// ```
-  static Transaction<String> stringTerminated(Stream<Uint8List> stream, Uint8List terminator) {
-    return Transaction<String>(stream, TerminatedStringTransformer.broadcast(terminator: terminator));
+  static Transaction<String> stringTerminated(
+      Stream<Uint8List> stream, Uint8List terminator,
+      {int maxLen: 1024, bool stripTerminator: true}) {
```
Comment (altera2015) [inline lib/transaction.dart:76]:
```suggestion
      {int maxLen = 1024, bool stripTerminator = true}) {
```

**Review by altera2015 — CHANGES_REQUESTED** [review]
fix some lint warnings

**Comment by EParisot** [comment]
sry about that , I my Java env is not up to date

**Pull request merged**